### PR TITLE
feat: relay integration

### PR DESCRIPTION
**Pull request by bellini666**
Relay integration ported from strawberry-django-plus, using the official strawberry relay integration

### CODE REVIEW

File: strawberry_django/fields/base.py [hunk 1]
```diff
@@ -43,11 +60,86 @@ def is_basic_field(self) -> bool:
         """
         return False
 
-    @property
-    def type(  # noqa: A003
+    @cached_property
+    def django_type(self) -> type[utils.WithStrawberryDjangoObjectDefinition] | None:
```
Comment (patrick91) [inline strawberry_django/fields/base.py:64]: nit: could we do this?

```suggestion
    def django_type(self) -> type[WithStrawberryDjangoObjectDefinition] | None:
```

File: strawberry_django/fields/base.py [hunk 2]
```diff
@@ -43,11 +60,86 @@ def is_basic_field(self) -> bool:
...
+        origin = self.type
+
+        tdef = get_object_definition(origin)
```
Comment (patrick91) [inline strawberry_django/fields/base.py:67]: nit, can we change tdef to type_definition or object_definition? :D

**Review by patrick91 — APPROVED** [review]
From a quick look it looks great!

File: strawberry_django/fields/base.py (hunk 1 above)
Comment (bellini666) [reply strawberry_django/fields/base.py]: Done! :)

File: strawberry_django/fields/base.py (hunk 2 above)
Comment (bellini666) [reply strawberry_django/fields/base.py]: Done! :)

**Pull request merged**